packages/extension-huggingface/src/lib/filters/tableQuestionAnswering.ts

```diff
@@ -3,82 +3,56 @@ import {
   InternalError,
   createFilterExtension,
 } from '@vulcan-sql/core';
-import axios, { AxiosError } from 'axios';
-import { convertToHuggingFaceTable } from '../utils';
-import { isArray } from 'class-validator';
-import { has } from 'lodash';
 
-type HuggingFaceOptions = {
-  accessToken: string;
-};
+import { convertToHuggingFaceTable, postRequest } from '../utils';
+import { has, isArray, isEmpty, omit } from 'lodash';
+import {
+  InferenceNLPOptions,
+  HuggingFaceOptions,
+  apiInferenceEndpoint,
+} from '../model';
 
-// More information described the options, see: https://huggingface.co/docs/api-inference/detailed_parameters#table-question-answering-task
+// More information described the options. See: https://huggingface.co/docs/api-inference/detailed_parameters#table-question-answering-task
 type TableQuestionAnsweringOptions = {
   inputs: {
     query: string;
     table: Record<string, string[]>;
   };
-  options: {
-    use_cache: boolean;
-    wait_for_model: boolean;
-  };
-};
-
-const request = async (url: string, data: any, token: string) => {
-  try {
-    const result = await axios.post(url, data, {
-      headers: { Authorization: `Bearer ${token}` },
-    });
-    return result.data;
-  } catch (error) {
-    const axiosError = error as AxiosError;
-    // https://axios-http.com/docs/handling_errors
-    // if response has error, throw the response error, or throw the request error
-    if (axiosError.response)
-      throw new Error(JSON.stringify(axiosError.response?.data));
-    throw new Error(axiosError.message);
-  }
+  options?: InferenceNLPOptions;
 };
 
-// default recommended model, see https://huggingface.co/docs/api-inference/detailed_parameters#table-question-answering-task
+/**
+ * Get table question answering url. Used recommend model be default value.
+ * See: https://huggingface.co/docs/api-inference/detailed_parameters#table-question-answering-task
+ * */
 const getUrl = (model = 'google/tapas-base-finetuned-wtq') =>
-  `https://api-inference.huggingface.co/models/${model}`;
+  `${apiInferenceEndpoint}/${model}`;
 
 export const TableQuestionAnsweringFilter: FunctionalFilter = async ({
   args,
   value,
   options,
 }) => {
-  if (!options || !(options as HuggingFaceOptions).accessToken)
-    throw new InternalError('please given access token');
+  const token = (options as HuggingFaceOptions)?.accessToken;
+  if (!token) throw new InternalError('please given access token');
 
   if (!isArray(value))
     throw new InternalError('Input value must be an array of object');
-
   if (!(typeof args === 'object') || !has(args, 'query'))
     throw new InternalError('Must provide "query" keyword argument');
-  if (!args['query'])
-    throw new InternalError('The "query" argument must have value');
```

Comment:
Curious about why we removed this query check.

Reply:
Thanks for asking the question. I added back the logistic for checking "query" has value or not with test cases in 4c445cdb341c70090eff5b0605deea13894876f9